### PR TITLE
For SG-5387: Adds support for Hiero/Nuke Studio clips.

### PR DESCRIPTION
**Pull request by thebeeland**
This is a cleaned-up implementation of client PR #10. It implements support for checking and updating clip paths that might be referencing published data.

### CODE REVIEW

File: hooks/tk-nuke_scene_operations.py [hunk 1]
```diff
@@ -39,10 +42,30 @@ def scan_scene(self):
...
+        # clips we need to be aware of that we might want to point to newer
+        # publishes.
+        if self.parent.studio_enabled or self.parent.hiero_enabled:
+            import hiero.core
+
+            for project in hiero.core.projects():
+                for clip in project.clipsBin().clips():
+                    files = clip.activeItem().mediaSource().fileinfos()
+                    for file in files:
+                        path = file.filename().replace("/", os.path.sep)
```
Comment (josh-t) [inline hooks/tk-nuke_scene_operations.py]: maybe use [`ShotgunPath.normalize`](http://developer.shotgunsoftware.com/tk-core/utils.html#sgtk.util.ShotgunPath.normalize) here?

File: hooks/tk-nuke_scene_operations.py [hunk 2]
```diff
@@ -53,6 +76,7 @@ def scan_scene(self):
             path = node.knob('file').value().replace("/", os.path.sep)
 
             reads.append( {"node": node_name, "type": "Read", "path": path})
+            logger.debug("inside my code, i can log like thiszzzzz")
```
Comment (josh-t) [inline hooks/tk-nuke_scene_operations.py]: remove?

File: hooks/tk-nuke_scene_operations.py [hunk 3]
```diff
@@ -82,19 +106,22 @@ def update(self, items):
...
-
         engine = self.parent.engine
 
         node_type_list = ["Read", "ReadGeo2", "Camera2"]
+
         for i in items:
             node_name = i["node"]
             node_type = i["type"]
-            new_path = i["path"]
+            new_path = i["path"].replace(os.path.sep, "/")
```
Comment (josh-t) [inline hooks/tk-nuke_scene_operations.py]: same comment regarding `ShotgunPath.normalize`

File: hooks/tk-nuke_scene_operations.py (hunk 2 above)
Comment (thebeeland) [reply hooks/tk-nuke_scene_operations.py]: Err...yeah...didn't catch that in the original PR.

File: hooks/tk-nuke_scene_operations.py (hunk 1 above)
Comment (thebeeland) [reply hooks/tk-nuke_scene_operations.py]: I'm going to stick with this to match other logic in this hook. It's old code, and I don't really want to rock the boat.

**Pull request merged**